### ISSUE TITLE
Test dependencies are not registered in gradle

### ISSUE THREAD

**Issue by matmannion** [scalacenter/bloop#808]
I have a 3-project Gradle build with 3 projects, each with their own main and test:

* common
* api
* web

`api` and `web` depend on `common` and gradlew bloopInstall handles this great, but for the test projects it incorrectly depends on `common` twice instead of `common` and `common-test`, e.g. this is from the generated `.bloop/api-test.json`:

```
  "name" : "api-test",
  ...
  "dependencies" : [
    "api",
    "common",
    "common"
  ],
```

**Comment by jvican** [comment]
I think the cause of this bug is described in https://github.com/scalacenter/bloop/issues/800, where we don't query transitively project dependencies. I'm closing in favor of the older bug report.

**Comment by Arthurm1** [comment]
@matmannion 
Do you mean your **api** project tests depend on the test source set of the **common** project?

If so, how do you specify that?  Something like...

**api** build.gradle:
```
dependencies {
  testCompile project(':common').sourceSets.test.output
}
```

or

**common** build.gradle:
```
configurations {
  tests {
    extendsFrom testCompile
  }
}

task testJar(type: Jar) {
    archiveBaseName = '???'
    from sourceSets.main.output
    include('somepath/**')
}

artifacts {
  tests testJar
}
```

**api** build.gradle:
```
dependencies {
  testCompile project( path: ':common', configuration: 'tests')
}
```

If it's the latter, can you supply samples of the relevant sections of both build files please?

**Comment by matmannion** [comment]
Hi @Arthurm1, it's a slight variation on the latter:

`api` build.gradle:

```
dependencies {
    compile project(':common')
    testCompile project(path: ':common', configuration: 'testArtifacts')
}
```

*common* build.gradle:

```
configurations {
    testArtifacts.extendsFrom testRuntime
}

artifacts {
    testArtifacts testJar
}
```

**Comment by halfninja** [comment]
The referenced PR fixes #800 but not this issue so should this be reopened to ensure it's still covered?

**Comment by jvican** [comment]
I think @Arthurm1 is testing that the test dependencies work in its PR #818. As this ticket is a subset of #800, I don't see a reason to reopen it. Let me know if I miss something

**Comment by halfninja** [comment]
Just because his PR description specifically says "Doesn't fix #808". Unless he's told you otherwise separately!

**Comment by jvican** [comment]
Right, I didn't notice that slight detail. Let me ask in the PR for more clarifications, thanks for shedding light on this issue @halfninja
